addons/html_builder/static/src/core/building_blocks/builder_number_input.js

```diff
@@ -108,12 +108,21 @@ export class BuilderNumberInput extends Component {
         displayValue = displayValue.replace(/,/g, ".");
         // Only accept 0-9, dot, - sign and space if multiple values are allowed
         if (this.props.composable) {
-            displayValue = displayValue.replace(/[^0-9.-\s]/g, "");
+            displayValue = displayValue
+                .trim()
+                .replace(/[^0-9.-\s]/g, "")
+                .replace(/(?<!^|\s)-/g, "");
         } else {
             displayValue = displayValue
                 .trim()
+                // Remove spaces are a "-" to accept "- 10" as "-10"
+                .replace("- ", "-")
```

Comment:
if you accept "- 10", you maybe want to accept "-     10" ? Maybe a question for the PO